source/22_test/index.rst

```diff
@@ -0,0 +1,155 @@
+* ``platformtest`` -- unit test of cross platform code, ensure that the core functions of DI engine can still operate normally on MacOS and Windows platforms.
+* ``benchmark`` -- performance test of algorithm or architecture, speed measurement is mainly carried out for relevant contents to ensure that its performance meets the requirements.
+
+
+
+How to Build Unit Test
+---------------------------------
+
+In DI engine, we use `pytest <https://docs.pytest.org/>`_ to build unit tests.
+
+For unit test writing, you can refer to the ``tests`` folder under the code path at all levels as a whole, such as `ding/envs/env_manager/tests <https://github.com/opendilab/DI-engine/tree/main/ding/envs/env_manager/tests>`_.
+
+
+Naming Conventions
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+For unit testing, we generally build it in the unit of class or function, and its name should meet certain specifications, specifically:
+
+* For the unit test of function form, the function name is required to be started with ``test_``.
+* For the unit test of class form, the class name is required to be started with ``Test``, and names of all methods used for testing should start with ``test_``.
+
+
+Assertions
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+In the test cases, we use ``assert`` (assertion) to check the prototype results. If the assertion is not true, very detailed information will be displayed, as shown in the following figure
+
+.. image:: pytest_assert.png
+    :scale: 55%
+    :align: center
+
+In addition, `` pytest`` also supports assertion of thrown exceptions, as shown below
+
+.. code:: python
+
+   import pytest
+
+   def test_zero_division():
+       with pytest.raises(ZeroDivisionError):
+           1 / 0
+
+In addition, for the test of real numbers, due to the storage principle of real numbers, it may lead to misjudgment caused by subtle errors. Therefore, the approximate function ``approx`` can be used for approximate judgment. It supports numeric type, ``list`` type, ``dict`` type and ``numpy`` type (``numpy. Ndarray``).
+
+.. image:: pytest_approx.png
+    :scale: 55%
+    :align: center
+
+
+
+Fixture and Conftest
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Fixture is a very important mechanism in ``pytest``. It can initialize the resources required for testing and pass them in as parameters of the test function for its usage. Not only that, but also the recovery of operation resources can be realized to ensure that the subsequent operation will not be affected. In addition, code reuse can be easily realized through the definition of scope.
+
+This `Chinese tutorial <https://www.cnblogs.com/linuxchao/p/linuxchao-pytest-fixture.html>`_ is written in great detail and can be used as a reference. In the existing code of DI-engine, you can refer to `ding/league/tests/test_player.py <https://github.com/opendilab/DI-engine/tree/main/ding/league/tests/test_player.py>`_.
```

Comment:
Find an english tutorial here, or just discard this part.

Reply:
Fixed.